tests/com.espressif.idf.ui.test/src/com/espressif/idf/ui/test/executable/cases/project/NewEspressifIDFProjectSBOMTest.java

```diff
@@ -0,0 +1,216 @@
+		Fixture.whenProjectIsBuiltUsingContextMenu();
+		Fixture.whenRunSBOMtool();
+		Fixture.thenCheckResultInConsole();
+	}
+
+	@Test
+	public void givenNewProjectCreatedBuiltWhenRunSBOMtoolRedirectOutputToFileThenVerifyConsoleAndCheckResultInFile()
+			throws Exception
+	{
+		Fixture.givenNewEspressifIDFProjectIsSelected("EspressIf", "Espressif IDF Project");
+		Fixture.givenProjectNameIs("NewSbomProjectThirdTest");
+		Fixture.whenNewProjectIsSelected();
+		Fixture.whenProjectIsBuiltUsingContextMenu();
+		Fixture.whenRunSBOMtoolUsingContextMenuRedirectOutputToFile();
+		Fixture.thenVerifyConsole();
+		Fixture.thenCheckResultInFile();
+	}
+
+	@Test
+	public void givenNewProjectCreatedBuiltWhenOpenSBOMtoolThenCheckPathValidation() throws Exception
+	{
+		Fixture.givenNewEspressifIDFProjectIsSelected("EspressIf", "Espressif IDF Project");
+		Fixture.givenProjectNameIs("NewSbomProjectFourthTest");
+		Fixture.whenNewProjectIsSelected();
+		Fixture.whenProjectIsBuiltUsingContextMenu();
+		Fixture.whenOpenSBOMtool();
+		Fixture.thenCheckPathValidation(Fixture.bot);
+	}
+
+	private static class Fixture
+	{
+		private static SWTWorkbenchBot bot;
+		private static String category;
+		private static String subCategory;
+		private static String projectName;
+
+		private static void loadEnv() throws Exception
+		{
+			bot = WorkBenchSWTBot.getBot();
+			EnvSetupOperations.setupEspressifEnv(bot);
+			bot.sleep(1000);
+		}
+
+		private static void givenNewEspressifIDFProjectIsSelected(String category, String subCategory)
+		{
+			Fixture.category = category;
+			Fixture.subCategory = subCategory;
+		}
+
+		private static void givenProjectNameIs(String projectName)
+		{
+			Fixture.projectName = projectName;
+		}
+
+		private static void whenNewProjectIsSelected() throws Exception
+		{
+			ProjectTestOperations.setupProject(projectName, category, subCategory, bot);
+		}
+
+		private static void whenProjectIsBuiltUsingContextMenu() throws IOException
+		{
+			ProjectTestOperations.buildProjectUsingContextMenu(projectName, bot);
+			ProjectTestOperations.waitForProjectBuild(bot);
+			TestWidgetWaitUtility.waitForOperationsInProgressToFinishAsync(bot);
+		}
+
+		private static void cleanTestEnv()
+		{
+			TestWidgetWaitUtility.waitForOperationsInProgressToFinishAsync(bot);
+			ProjectTestOperations.closeAllProjects(bot);
+			ProjectTestOperations.deleteAllProjects(bot);
+		}
+
+		private static void whenOpenSBOMtool() throws IOException
+		{
+			ProjectTestOperations.launchCommandUsingContextMenu(projectName, bot, "SBOM Tool");
+		}
+
+		private static void thenRunOkbuttonDisabled(SWTWorkbenchBot bot) throws IOException
+		{
+			SWTBotButton okButton = bot.button("OK");
+			assertTrue(!okButton.isEnabled());
+			bot.checkBox("Redirect output to the file").click();
+			assertTrue(!okButton.isEnabled());
+			bot.checkBox("Redirect output to the file").click();
+			assertTrue(!okButton.isEnabled());
+			bot.button("Cancel").click();
+		}
```

Comment:
again in this then function we must only be using asserts not doing operations on anything
If you need to verify this that the button remains disabled add another when method and better to have a separate test for this as well.